javalanglib/src/main/scala/java/lang/ClassValue.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.lang
+
+import java.util.HashMap
+
+import scala.scalajs.js
+import scala.scalajs.runtime.linkingInfo
+
+abstract class ClassValue[T] protected () {
+  import ClassValue._
+
+  private val jsMap: AnyRef = {
```

Comment:
Can't this be a `js.Dynamic`? IIUC, `js.Dynamic` would be erased to `any` in the IR, so it's not a problem if we use it. Or am I missing something?

Reply:
Yes, you're right. That works. I don't know why I thought otherwise.

Reply:
Actually, looking at the IR cleaner: It seems good enough that you can actually use the full facade of `js.Map` here, no?

Reply:
Ok, seems we don't have a public facade for map, so we indeed cannot use it. Consider making a private one though. It might avoid a lot of casts (depends a bit on the outcome of the `Sentinel` discussion as well, because with the sentinel, we cannot type it properly anyways).

Reply:
I had forgotten how smart our IR cleaner actually is. I was convinced that declaring any facade inside javalanglib was forbidden, but in fact it is capable of erasing them.

We have `js.Map`, but most methods are hidden in favor of the Scala collections methods. I have added direct accessors in `Utils`.